src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1388,36 @@ public S(int arg)
 ");
             var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
             comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.NotEmpty(comp.GetTypeByMetadataName("S").GetMembers(GeneratedNames.MakeBackingFieldName("P")));
+            comp.VerifyDiagnostics();
 
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void SpeculativeSemanticModelTest()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    public int P { get => 0; }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers(GeneratedNames.MakeBackingFieldName("P")));
             comp.VerifyDiagnostics();
 
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken));
+
+            var model = comp.GetSemanticModel(comp.SyntaxTrees[0]);
+            var identifier = SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(leading: default, contextualKind: SyntaxKind.FieldKeyword, text: "field", valueText: "field", trailing: default));
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, identifier, out var speculativeModel);
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetFieldsToEmit());
+
+            var fieldKeywordTypeInfo = speculativeModel.GetTypeInfo(identifier);
+            Assert.Equal(SpecialType.System_Int32, fieldKeywordTypeInfo.Type.SpecialType);
```

Comment:
Can you please verify if this is expected to work? Currently it doesn't work because (as far as I understand), before speculation, we already marked the backing field as computed.

Reply:
> Can you please verify if this is expected to work? 

Define "to work". 
We are in the investigation territory here. Reviewers should be able to see the current behavior. Meaning, the tests should pass, they should reflect the current behavior. Then we can examine if the behavior matches our expectations.